### ISSUE TITLE
The Out-of-Order Corollary

### ISSUE THREAD

**Issue by Zamiell** [hanabi/hanabi.github.io#620]
This is not a new convention. It is the way that we have already been playing for the last 4 years.

### The Out-of-Order Corollary

- First, see the section on the *[Out-of-Order Play Clue](#the-out-of-order-play-clue-triple-o--ooo)*.
- As specified in the sections above, if Alice gives a clue to Cathy touching the next playable card, but the focus is wrong, Bob would normally give a OOO *Fix Clue* instead of blind-playing a card.
- Thus, if Alice gives a clue to Cathy, and Bob does blind-play a card, then by default, Cathy can mark all the cards that were touched with a note of not being the next playable card.
- For example, in a 3-player game:
  - Red 2 is played on the stacks.
  - Alice clues red to Cathy, touching two cards on slot 1 and slot 2.
  - Bob blind-plays a blue 1 and it successfully plays on the stacks.
  - Cathy marks her slot 1 card as the red 4. (Cathy knows that Alice performed a *Bluff* and slot 1 was the focus of the clue.)
  - Cathy also marks her slot 2 card as the red 5. (Cathy knows that if it was a red 3, then Bob would be forced to give a *Fix Clue*.)

**Comment by sjdrodge** [comment]
> This is not a new convention. It is the way that we have already been playing for the last 4 years.

This may be the way you have been playing, but it is not the way everybody has been playing.

This strikes me as part of a series of changes designed to wring all of the context and judgement out of our convention system.

Currently, it's quite common that when r3 is on the stacks and Cathy holds r4 g3 y2 r5, Alice clues red to Cathy and Bob blind-plays (thinking he's providing r4). Bob makes this assumption because it would be _extremely weird_ for Alice to clue red instead of 4 or 5 when both of those clues are unblocked. Furthermore, if Bob ends up playing something other than r4, Cathy should not go on to assume that slot 1 is trash. Cathy is still supposed to apply Good Touch Principle.

It is not possible to enumerate all the scenarios where a lie should be deemed "good enough," and we should not try to do so. Dictating that players should always assume a lie (complete w/ fix clue) in possible OOO scenarios is simple, but is seriously lacking in other merits.

**Comment by Zamiell** [comment]
as stated in the context section, the document just specifies the default or most common case.
it doesn't dictate to players to not use context.
if players think that a lie is good enough, then they should disregard the corollary, just like they would any other section in the document

**Comment by sjdrodge** [comment]
> as stated in the context section, the document just specifies the default or most common case.
> it doesn't dictate to players to not use context.
> if players think that a lie is good enough, then they should disregard the corollary, just like they would any other section in the document

You misunderstand. The corollary you've written instructs players to assume that a lie has occurred that they must now fix _even when they can see that the truth would accomplish the same task_. This runs counter to the way we actually play, where Bob instead only assumes a lie if the same thing couldn't have been accomplished with the truth.

**Comment by Zamiell** [comment]
> even when they can see that the truth would accomplish the same task

i've written no such thing

> This runs counter to the way we actually play

ime, the way we play is that no-one would bluff this way, because it is undefined and confusing

**Comment by sjdrodge** [comment]
> i've written no such thing

In the example given in my first message in this thread. 4 or 5 followed by reds plainly works, and the corollary that you have written instructs players that as Bob, if Alice gives reds instead of either of those two alternatives, they should give a fix clue instead of blind-playing.

In other words, even though Bob can see that the truth works w/ the same efficiency as the OOO lie, but you've told players `As specified in the sections above, if Alice gives a clue to Cathy touching the next playable card, but the focus is wrong, Bob must give a Fix Clue instead of blind-playing a card.`

I am eager to hear how I've misrepresented what you've written.

**Comment by Zamiell** [comment]
if alice has a free choice between a red color clue and a number 4 clue, and chooses to clue red instead of a number 4 clue, then that is an undefined move, and bob is free to interpret that however he wishes. the document doesn't care and neither do i. if you want to propose that as a move then create a new issue.

the words "bob must give a fix clue" does not apply to undefined situations

**Comment by sjdrodge** [comment]
If this change to the document is made as-is. The situation becomes defined. It becomes defined as Bob must give Fix Clue and should not blind-play. That is a bad definition. Hence the problem!

**Comment by Zamiell** [comment]
> It becomes defined as Bob must give Fix Clue and should not blind-play.

bob must give a fix clue and should not blind-play, barring any exceptional circumstances, just like every other section in the document

> Cathy should not go on to assume that slot 1 is trash. Cathy is still supposed to apply Good Touch Principle

by default, cathy should go on to assume this.
cathy should only break from the default interpretation if there is some other piece of context.

> it's quite common

it is not common. link 3 games where it has happened

**Comment by sjdrodge** [comment]
I invite you to post three replays where Alice gives a clue similar to the one in my example and intends it as OOO. OOO Finesses have very little to do with the problems in the proposed verbiage.

It may be true to you that it's a highly exceptional circumstance that need not be stated that OOO isn't intended when the truth would work just as well, but the average reader of this proposed section will not come away with that impression.

If it is not explained to the reader that they should always consider whether the truth would work just as well as the lie before assuming a lie, then they will simply take the statement `if Alice gives a clue to Cathy touching the next playable card, but the focus is wrong, Bob must give a Fix Clue instead of blind-playing a card` at face value.

Also, as an aside, you gotta stop with this whole gatekeeping thing. If people are reading/contributing to the document and playing with the conventions, their input/experience is valuable. It doesn't matter whether they play with you or your friends as often as you'd like.

**Comment by Zamiell** [comment]
it doesn't need to be explained to the reader because it isn't the default case. if it were frequent enough, then sure, we could throw in a bullet point. im not going to throw in a bullet point though for a move that doesn't exist.

**Comment by pianoblook** [comment]
> ime, the way we play is that no-one would bluff this way, because it is undefined and confusing

ime no one would give a play-clue this way because it's undefined, confusing, *and* terrible. At least if Bob blind-plays it becomes undefined, confusing, and efficient 🤷

**Comment by pianoblook** [comment]
I'm imaging the Venn Diagram of (A) Players who know about the existence and nuances of _OOO_ and (B) Players who would ever in their right mind decide to give such a clue in such a situation, and the overlap is very clearly zero.

**Comment by jack67889** [comment]
Let's go back to the original question, because imo it's already defined in the doc:

> Level 23. Red 3 on stack. Alice clues Red to Cathy touching Red 5 on chop and Red 4 on slot 1. Bob responds by playing slot 1 as a known bluff. Cathy plays Red 4 from slot 1 since Bob’s blind play let her know that her red cards are in different order than she thought.

The doc has a new section on OOO play clues, which says:

> Sometimes, a player will have two playable cards in their hand of the same suit, but they will be "blocked", meaning that both a color clue and a number clue will not be able to focus the first playable card.

That situation doesn't apply in this case, so by default it's a finesse/known bluff over an OOO play clue according to the doc, imho.

Moreover I think the main confusion/disagreement arises because the (older) section on OOO finesses says:

> The previously mentioned Out-of-Order Finesse is a specific example of a Finesse with a Lie Component. (…)
Importantly, Finesses with a Lie Component are **only** allowed if there is not an alternative line that would "get" the same number of cards without any lies. Thus, if you are looking to do a Finesse with a Lie Component, make sure you carefully consider the efficiency of all of the possible alternatives.

To reflect the way the group has been playing I think it makes most sense that this effiency requirement extents to OOO play clues. So for the example it makes sense (in case there are no blocking cards) that you are still allowed to give an OOO play clue if Bob can give an efficient fix.

**Comment by Zamiell** [comment]
> To reflect the way the group has been playing I think it makes most sense that this efficiency requirement extents to OOO play clues.

you are prescribing behavior for bob. if you want to do that, i think you should open a separate issue, because my thread is about clarifying what cathy should think, not bob. cathy has no idea whether her hand has a blocking card or not. and by default, she assumes that she does have a blocking card.

**Comment by Zamiell** [comment]
> At least if Bob blind-plays it becomes undefined, confusing, and efficient

it sounds like you are prescribing behavior for bob. if you want to do that, i think you should open a separate issue, because my thread is about clarifying what cathy should think, not bob.

**Comment by jakestiles** [comment]
I think this new corollary is bad and it is not the way I play. In the future Zamiel, I think you should avoid claiming that everyone plays the way you play. I agree with Most of Stephen’s and timotree’s and Valletta’s points in the Convention-questions thread.

**Comment by Zamiell** [comment]
>  I think you should avoid claiming that everyone plays the way you play.

i've not done so

**Comment by jack67889** [comment]
> you are prescribing behavior for bob. if you want to do that, i think you should open a separate issue, because my thread is about clarifying what cathy should think, not bob. in the case where cathy gets a OOO fix clue, cathy has no idea whether her hand has a blocking card or not. and by default, she should assume that she does have a blocking card.

You are prescribing behaviour for Bob and behaviour for Cathy based on the behaviour of Bob. I'm pointing out the doc currently prescribes different behaviour for both Bob and Cathy.

Furthermore if the focus of the clue is trash it would be an UTD/UTC according to the doc, so that can't be the reason for Bob to play finesse.

> For an Unknown Trash Discharge:
> - The empathy on the clued card contains one or more possibilities that are useful and currently unplayable.
> - Thus, something extra is needed to prove it is trash.
> Other than this, don't use more complicated factors to prefer a Trash Bluff over an Unknown Trash Discharge (like Bob having to wait on some other card).

**Comment by jakestiles** [comment]
- “that is the way that we play...we = the hyphenated group”
- “This is not a new convention. It is the way that we have already been playing for the last 4 years.”

These are the comments I take issue with. So far you are outnumbered on this topic. This topic is either solidly disputed or more people disagree than agree with you.

**Comment by pianoblook** [comment]
> it sounds like you are prescribing behavior for bob. if you want to do that, i think you should open a separate issue, because my thread is about clarifying what cathy should think, not bob.

I'm confused. This post explicitly says that in situations like this, "Bob must give a Fix Clue instead of blind-playing a card." That sounds like you're prescribing behavior for Bob, no? And my post, inversely, is disagreeing that such a situation should be necessarily prescribed. 

As myself and others have argued, it seems extremely reasonable for Bob - at least in a large portion of possible scenarios - to blind-play instead of giving a Fix clue.

**Therefore**, if Cathy is a good Hanabi player and realizes that there are many realistic scenarios in which Bob would be blind-playing instead of _Fixing_, it would be preposterous to assume she should automatically mark her other card(s) as Trash. That sounds like you're prescribing bad behavior to Cathy

**Comment by Zamiell** [comment]
jake, in this context the word "we" does not mean every single individual person inside of the hyphen-ated group, of which there are hundreds of people. i can't claim to read everyone's minds or know exactly how everyone individual is playing. however, i am in a unique position to have player with a greater portion of these people than anybody else. when I saw we in this context I mean "most people". hopefully that makes sense

**Comment by Zamiell** [comment]
i updated the wording slightly:

```
- As specified in the sections above, if Alice gives a clue to Cathy touching the next playable card, but the focus is wrong, Bob would normally give a OOO *Fix Clue* instead of blind-playing a card.
- Thus, if Alice gives a clue to Cathy, and Bob does blind-play a card, then by default, Cathy can mark all the cards that were touched with a note of not being the next playable card.
```

> You are prescribing behaviour for Bob and behaviour for Cathy based on the behaviour of Bob.

based on this new edit, no, i am not prescribing behavior for bob.
rather, i am making a descriptive claim about what bob does >50% of the time when this situation occurs, from an entirely non-normative standpoint, from cathy's perspective.

> I'm pointing out the doc currently prescribes different behaviour for both Bob and Cathy.

i do not think that the doc prescribes different behavior.
for discussion purposes, let's call the move that Stephen is potentially proposing as a *Faulty Bluff*.
the doc only defines what OOO, it does not define what a *Faulty Bluff*.
therefore from Cathy's perspective, she should only assume that defined moves have happened in her game.

**Comment by Zamiell** [comment]
> This post explicitly says that in situations like this, "Bob must give a Fix Clue instead of blind-playing a card." That sounds like you're prescribing behavior for Bob, no?

no, i am not prescribing behavior for bob. see my edit

**Comment by pianoblook** [comment]
Here's what I'd expect to see if I came across this example in the doc:

> For example, in a 3-player game:
> - Red 2 is played on the stacks.
> - Alice clues red to Cathy, touching two cards on slot 1 and slot 2.
> - Bob blind-plays an unrelated card.
> - Cathy marks her slot 1 card as the red 4. (Cathy knows that Alice performed a Bluff.)
- Cathy also marks her slot 2 card as red 3 or red 5. (Cathy in general should know that she shouldn't be prescribed to mark her card as not being red 3, since level 19 Hanabi players are well-versed in conventions such as _Context_ and _Finesses with a Lie Component_. 

I really don't understand why on earth Cathy should **always** mark slot 2 as never being red 3 - again, that feels like a bad convention.

**Comment by pianoblook** [comment]
> cathy doesn't always mark her slot 2 as being red 3. however, by default she does (in the absence of any context), because OOO exists, and _Faulty Bluffs_ don't exist

Your example clearly says she should mark it as being red 5 exactly:
> Cathy also marks her slot 2 card as the red 5. **(Cathy knows that if it was a red 3, then Bob would be forced to give a Fix Clue.)**

We're saying that this statement is false and should not be the general default

**Comment by Zamiell** [comment]
> This post clearly says she should mark it as being red 5 exactly:

i meant that she doesn't *always* mark her slot 2 card as being red 5.

**Comment by Zamiell** [comment]
> We're saying that this statement is false and should not be the general default

it should be the general default, because OOO exists, and Faulty Bluffs don't exist.

as explained in the section on context, the doc only prescribes the default thing to do when there is no particular context involved

**Comment by jakestiles** [comment]
Sure OOO exists but it does not declare itself to be mandatory or the only way to get out of order cards. In fact, its 2nd bullet point says OOO is Just one such way to get the cards. 

I disagree that the doc currently prescribes this interpretation and I disagree that it should. 

In my opinion, Bob is currently allowed and should be allowed to rule out OOO if the fix isn’t good or he has no clues to fix with. In my opinion, Cathy shouldn’t assume good touch principle has been violated just because Bob blind plays.

**Comment by jack67889** [comment]
> however, by default she does (in the absence of any context), because OOO exists, and Faulty Bluffs don't exist.

I can repeat myself, but see my first post. Doc explicitly defines this to be not an OOO play clue, so it's definitely a finesse/bluff by default.

Again, I think a new section to clarify OOO play clues shouldn't change how the group should treat OOO clues in general. The efficiency requirement has been there, so that's how the group has been playing. (Didn't see an argument yet why this should be changed.)

**Comment by timotree3** [comment]
I don't think "Faulty Bluffs" need to be defined separately to exist conventionally. The definition of a bluff is enough

**Comment by Zamiell** [comment]
> In my opinion, Bob is currently allowed and should be allowed to rule out OOO if the fix isn’t good or he has no clues to fix with. I

that's fine. again, this thread isn't about bob, it is about cathy. if you want to prescribe something new for bob, please open a new issue!

> Doc explicitly defines this to be not an OOO play clue, so it's definitely a finesse/bluff by default.

i dont think so. if it isn't OOO, then it is undefined. if bob wants to interpret it as a finesse, then that's certainly a reasonable interpretation, but that isn't defined. and you shouldn't claim that it is.

> Again, I think a new section to clarify OOO play clues shouldn't change how the group should treat OOO clues in general.

this new section does not change how the group treats OOO clues.
what it does do is affect situations where players give undefined moves.
i am not concerned with undefined moves.

> I don't think "Faulty Bluffs" need to be defined separately to exist conventionally. The definition of a bluff is enough

tim, the way that we do things is that we give everything a different name. this helps for both comprehension and for discussion. why do you think we distinguish a Hard 3 Bluff from a normal 3 Bluff, even when they are extremely similar?

i dont think that you should blind-play in an undefined way and claim that "it's just a bluff".
if it is a common move, then come up with a name for it, define it, and put it in the document!

**Comment by pianoblook** [comment]
mandating something weird like this and telling us to open some *other* proposal just to conform with this weird mandate feels unnecessarily complicated and cluttered.

If you agree that Cathy shouldn't necessarily believe anything in specific about her slot 2 card - and instead should trust context etc - then why on earth do we need this extra Corollary?

**Comment by Zamiell** [comment]
> If you agree that Cathy shouldn't necessarily believe anything in specific about her slot 2 card

i don't agree with that.
my position is that by default, in most situations, cathy should write red 5 on her slot 2 card, but not in every situation.

**Comment by jack67889** [comment]
> this new section does not change how the group treats OOO clues.
what it does do is affect situations where players give undefined moves.
i am not concerned with undefined moves in this thread.

I see this differently. To me the way the group has been playing is OOO clues/finesses with a lie component in general can be confusing and are not fully defined. So only do them if you are sure they meet the efficiency requirement. In the original example the only way the OOO clue can meet the efficiency requirement is if Bob blind-plays. In my eyes this new definition you are proposing removes the efficiency requirement for OOO play clues and forces Bob to fix instead (thus Bob should assume Alice gives a confusing clue for no reason), doesn’t seem valuable enough to me. (And yes, you are still prescribing behaviour for Bob, since he has to react to the clue.)

**Comment by Zamiell** [comment]
> mandating something weird like this

piano, I would not frame it that way. the way that I would frame it is like this:

piano - "Hey Zamiel, I read your new issue thread and I am very surprised by it. In the past I have performed one or more *Faulty Bluffs* , and they have worked great! If your new convention is accepted, then *Faulty Bluffs* would not work anymore. Thus, this corollary seems problematic, so I think it should be modified and/or deleted as to not interfere with *Faulty Bluffs*."

Zamiel - "Um, piano, the *Faulty Bluff* doesn't actually exist. Show me the section in the document where it says that a *Faulty Bluff* legal."

piano - [looks through the document] "Oh, you're right. That's strange, *Faulty Bluffs* are not the document. That's weird, as it seems like a simple enough move. As I've said, I've done them before and they work great! And if I've done them, probably other people have done them too. So for the purposes of this thread, we should probably consider them as existing."

Zamiel - "If they are a common move, then lets create an issue for them and create a section for them in the doc. When the new section gets added, I'll completely delete the section called *The Out-of-Order Corollary*, since it will no longer apply. Furthermore, I'll add a bullet point to the OOO section that says: `Remember that sometimes, clues that look like OOO are not OOO at all and are instead *Faulty Bluffs*. Don't get them confused!`

Zamiel - "However, as it stands right now, that section doesn't exist. So no, for the purposes of this thread, we should not consider them as existing. For the purposes of this thread, *The Out-of-Order Corollary* follows from the existing conventions as written in a purely tautological manner. Meaning that this isn't "my" convention - it just logically follows from what we have already written."

**Comment by Zamiell** [comment]
> So only do them if you are sure they meet the efficiency requirement.

no. we commonly do OOO as a 2-for-2, so your post does not apply. if the doc is unclear on this please submit a pr

**Comment by jack67889** [comment]
Please provide an explanation how this applies to the original example without blocking cards.

**Comment by Zamiell** [comment]
to clarify, it almost always applies when there are not blocking cards.
cathy can't tell the difference, so she will assume it is the "almost always" part.

**Comment by jack67889** [comment]
That's not an answer to my question. So I can't comprehend your statements.

**Comment by Zamiell** [comment]
what part do you not understand

**Comment by jack67889** [comment]
> Please provide an explanation how this applies to the original example without blocking cards.

**Comment by Zamiell** [comment]
what original example?

**Comment by jack67889** [comment]
>Level 23. Red 3 on stack. Alice clues Red to Cathy touching Red 5 on chop and Red 4 on slot 1. Bob responds by playing slot 1 as a known bluff. Cathy plays Red 4 from slot 1 since Bob’s blind play let her know that her red cards are in different order than she thought.

**Comment by Zamiell** [comment]
that example describes undefined behavior, which this thread isn't concerned with. making undefined moves defined is for a new thread

**Comment by jack67889** [comment]
> I see this differently. To me the way the group has been playing is OOO clues/finesses with a lie component in general can be confusing and are not fully defined. So only do them if you are sure they meet the efficiency requirement. In the original example the only way the OOO clue can meet the efficiency requirement is if Bob blind-plays. In my eyes this new definition you are proposing removes the efficiency requirement for OOO play clues and forces Bob to fix instead (thus Bob should assume Alice gives a confusing clue for no reason), doesn’t seem valuable enough to me. (And yes, you are still prescribing behaviour for Bob, since he has to react to the clue.)

You try to define it as far as I understand. Otherwise I don't understand the need for this section.

**Comment by Zamiell** [comment]
nope, i am not defining anything new, only clarifying a logical consequence of the existing conventions as written

**Comment by jack67889** [comment]
That's not true, you just defined what Cathy should write on her cards when Bob blind-plays.

**Comment by Zamiell** [comment]
if someone reads the document very closely, and does not take into account undefined moves, then Cathy should write those notes on her cards regardless of whether or not the *The Out-of-Order Corollary* exists in the document

**Comment by florian-5f3759df** [comment]
> in other words, bob is not allowed to play slot 1 as a known bluff, so the example is nonsensical

Why is it a _known_ bluff? Bob is thinking that he's playing the other copy of red 4, it originally looks like a normal finesse I've seen before.

Only after he plays a card that's not red 4, it becomes a bluff (and then perhaps good touch applies to Cathy's slot 1 or not, I guess, depending on the result of this discussion).

**Comment by jack67889** [comment]
Which convention would make Bob blind-play finesse in the example of Jake and what would Cathy write on her cards?

**Comment by Zamiell** [comment]
> Which convention would make Bob blind-play finesse in the example of Jake

a finesse would, so cathy should write r5 on chop and kt as the other card

**Comment by jack67889** [comment]
So Bob should think it's a finesse and he should not fix if he can see the clue could be given without an OOO play clue for the same efficiency?

**Comment by Zamiell** [comment]
> Why is it a known bluff? Bob is thinking that he's playing the other copy of red 4, it originally looks like a normal finesse I've seen before.
> Only after he plays a card that's not red 4, it becomes a bluff (and then perhaps good touch applies to Cathy's slot 1 or not, I guess, depending on the result of this discussion).

we covered that already in the convention-questions channel, but to reiterate:
- in most games, there is a rainbow suit, so bob wouldn't typically be able to play r4 because then the other copy of r4 would be marked as a rainbow card, which would be a lie.
- in the end-game, this is no longer an issue, so a situation where Alice has *Free Choice* would result in Bob playing r4 as an *End-Game Distribution Finesse*.
- the end game is the most common situation where bob would play r4, but it could also happen in e.g. mid-game where there is no rainbow suit and Alice clearly has a *Free Choice*

however, notice that the *End-Game Distribution Finesse* does not conflict with *The Out-of-Order Corollary*.
in this situation, it is of no consequence whether or not alice marks her r4 as r1/r2/r3 or r1/r2/r3/r4, because it would be trash regardless.

**Comment by pianoblook** [comment]
There are plenty of undefined situations that involve a >50% correlation between a certain action and a certain underlying state regarding your hand. 

To throw a random example out there, let's say Alice _Chop Moves_ a red 2 in Cathy's hand while there are other clues to be given. Playing as Bob, from my experience I will now lightly assume that I probably don't have a r1 in my hand. At least for the sake of my argument, let's say that is probably true around 80% of the time.

Does that mean we should write up and add a _Chop Move Corollary_ prescribing Bob to write _Negative Elimination Notes_ for red 1 across his hand because of this action? I certainly hope not, because there's zero benefit. 

But uh oh, "the job of the doc is to describe the most common situations that come up, so I have typed up this section" - so I guess we need that Chop Move Corollary too, huh?

**Comment by jack67889** [comment]
I'm sorry but I try to grasp your statements, this example resulted in this issue and you are describing in the issue what Bob should do by default. I didn't understand yet what value you see in this section. And why would it be contrived??

**Comment by Zamiell** [comment]
> I'm sorry but I try to grasp your statements, this example resulted in this issue and you are describing in the issue what Bob should do by default. I didn't understand yet what value you see in this section.

no problem.
the value I see in the section is it clarifies what the common situation is for Cathy, after a clue like this happens:
https://hanabi.github.io/docs/level-12/#example-1---color-disconnect-valid
(but with the modification of where Cathy has two or more cards touched instead of one)

**Comment by Zamiell** [comment]
> Does that mean we should write up and add a Chop Move Corollary prescribing Bob to write Negative Elimination Notes for red 1 across his hand because of this action? I certainly hope not, because there's zero benefit.

this is a good point.
some sections in the doc describe "conventions" (or "moves", if you prefer).
other sections in the doc are "clarifications" on how conventions interact.

if a section called *Chop Move Corollary* did exist, it would be a "clarification" section.
but the clarification in this section would not be very useful:
1) the situation happens a medium-ish amount of the time (not super frequently though)
2) the situation is super obvious, you probably don't need a section telling you this
3) the situation "is only true" 80% of the time, and that isn't a very high number

based on these three points, the section would feel more like "bloat".
so i agree with you that a section called the *Chop Move Corollary* isn't necessary.

on the other hand, *The Out-of-Order Corollary* describes:
1) a situation that happens extremely frequently (after every single *Bluff* that touches two or more cards)
2) a situation that is not at all obvious, at least based on how much discussion has been around this in the past few days
3) a situation "is only true" 99% of the time (because for the purposes of this thread we must not consider undefined moves; see my previous reply)

thus, the calculus is different, and I assert that the section provides does provide value

**Comment by sjdrodge** [comment]
> that's fine. again, this thread isn't about bob, it is about cathy. if you want to prescribe something new for bob, please open a new issue!
 
You keep saying this, but other participants in the thread keep pointing out that you're prescribing behavior for Bob by saying Bob ought to give a Fix Clue instead of blind-playing in the example scenario. The reality is that Bob plays expecting to play r4.

In games, I see people get Bob's r4 in exactly this way with some frequency, but I very rarely see anyone use this well-defined behavior to bluff Bob. Perhaps that is where our disconnect is, if your comments about frequency and well-definedness are specifically about the bluff case.

In my mind, the major important point is that we should not add language to the document that would suggest to anyone that it's unusual to get Bob to play r4 in this fashion, or that would suggest that Bob should give a fix instead of blind-playing. That would be a major change to the way we have historically played, and also rather bad.

**Comment by sjdrodge** [comment]
> > mandating something weird like this
> 
> piano, I would not frame it that way. the way that I would frame it is like this:
> 
> piano - "Hey Zamiel, I read your new issue thread and I am very surprised by it. In the past, I have performed one or more _Faulty Bluffs_, and they have worked great! If your new convention is accepted, then _Faulty Bluffs_ would not work anymore. Thus, this convention seems problematic, so I think it should be modified and/or deleted as to not interfere with _Faulty Bluffs_."
> 
> Zamiel - "Um, piano, the _Faulty Bluff_ doesn't actually exist. Show me the section in the document where it says that a _Faulty Bluff_ legal."
> 
> piano - [looks through the document] "Oh, you're right. That's strange, _Faulty Bluffs_ are not in the document. That's weird, as it seems like a simple enough move. As I've said, I've done them before and they work great! And if I've done them, probably other people have done them too. So for the purposes of this thread, we should probably consider them as existing."
> 
> Zamiel - "If _Faulty Bluffs_ are a common move, then lets create an issue for them and create a section for them in the doc. When the new section gets added, I'll completely delete the section called _The Out-of-Order Corollary_, since it will no longer apply. Furthermore, I'll add a bullet point to the OOO section that says: `Remember that sometimes, clues that look like OOO are not OOO at all and are instead *Faulty Bluffs*. Don't get them confused!`
> 
> Zamiel - "However, as it stands right now, that section doesn't exist. So no, for the purposes of this thread, we should not consider them as existing. For the purposes of this thread, _The Out-of-Order Corollary_ follows from the existing conventions as written in a purely logical manner. Meaning that this isn't "my" convention - it just logically follows from what we have already written. Furthermore, for the purposes of this thread, we must not consider what "rogue" players are deciding to do on their own separate from the official moves and the official framework."

If this is really how you view this thread, what you're saying makes a lot of sense. To the rest of us it looks a little bit more like this:

Zamiel: "I'm going to add some language to the OOO section that says Bob should never blind-play from OOO-looking play clues and should instead always fix them."
Everyone: "WTF? why?"
Zamiel: "This is how we already play, if you want Bob to blind-play in these scenarios, you have to open an issue."
Everyone: "Wait what? We all give true finesses in this way with some frequency, and such behavior is supported by the current document, let's not change this, please"
Zamiel: "This change that says Bob should fix doesn't actually mean Bob should fix, open an issue if you want to talk about Bob."
Everyone: "???????????????????????????"

**Comment by Zamiell** [comment]
> In my mind, the major important point is that we should not add language to the document that would suggest to anyone that it's unusual to get Bob to play r4 in this fashion, or that would suggest that Bob should give a fix instead of blind-playing.

i agree that this section should not interfere with Bob assuming an *End-Game Distribution Finesse*.

as I've stated previously, I believe that this section is only describing what cathy should think - bob isn't involved.
if you want to suggest some edits to make that more explicitly, then feel free.

> I'm going to add some language to the OOO section that says Bob should never blind-play from OOO-looking play clues and should instead always fix them."

as I've stated previously, I believe that this section is only describing what cathy should think - bob isn't involved.
if you want to suggest some edits to make that more explicitly, then feel free.

do I sound like a broken record yet?

> Zamiel: "This change that says Bob should fix doesn't actually mean Bob should fix, open an issue if you want to talk about Bob."

please show me where it says that bob should fix.

**Comment by sjdrodge** [comment]
> i agree that this section should not interfere with Bob assuming an _End-Game Distribution Finesse_.

The end-game has nothing to do with it. This move can be performed at any point in the game.

> do I sound like a broken record yet?

Yes, please consider the possibility that it's not everybody else in this thread who is going crazy.
 
> > Zamiel: "This change that says Bob should fix doesn't actually mean Bob should fix, open an issue if you want to talk about Bob."
> 
> please show me where it says that bob should fix.

Both myself and others have already done this several times in the above thread by quoting your proposed change directly.

At this point I think it's unlikely that you'll understand the majority position, so you'll simply have to accept that you're outvoted here.

**Comment by Zamiell** [comment]
> The end-game has nothing to do with it. This move can be performed at any point in the game.

only partially agree, read my response to floriman

> quoting your proposed change directly

i just scrolled through the issue briefly and it looks like you haven't quoted any of the section?

**Comment by sjdrodge** [comment]
> i just scrolled through the issue briefly and it looks like you haven't quoted any of the section?

https://github.com/hanabi/hanabi.github.io/issues/620#issuecomment-804662503

**Comment by Zamiell** [comment]
what you've quoted is out of date and does not apply

**Comment by sjdrodge** [comment]
Yes, I haven't quoted your changed proposal, but others have.

**Comment by Zamiell** [comment]
please re-read the thread from start to finish.
i maintain that nothing in the *The Out-of-Order Corollary* section says that bob should give a fix clue in the hypothetical where it looks like an *End-Game Distribution Finesse* is happening.
if you think it does, please suggest an edit or provide a pull request

**Comment by 0nen** [comment]
I agree with the majority here.

**Comment by Zamiell** [comment]
""As specified in the sections above, if Alice gives a clue to Cathy touching the next playable card, but the focus is wrong, Bob would normally give a OOO Fix Clue instead of blind-playing a card."

this sentence is meant to be a descriptive statement, not a prescriptive one.

**Comment by Zamiell** [comment]
i.e. "here's a thing that has happened, now what should cathy thing?"

**Comment by pianoblook** [comment]
> Thus, if Alice gives a clue to Cathy, and Bob does blind-play a card, then by default, Cathy can mark all the cards that were touched with a note of not being the next playable card.

I maintain that this sentence is just patently false. In what world does this "default assumption" actually exist? In what world does this assumption have positive value to prescribe to Cathy? This just seems bad, and wrong, and not helpful under any circumstance

**Comment by pianoblook** [comment]
but at this point I don't think there's anything else we can say, other than a strong Downvote for this proposal

**Comment by Zamiell** [comment]
> In what world does this "default assumption" actually exist?

- For example, in a 3-player game:
  - Red 2 is played on the stacks.
  - Alice clues red to Cathy, touching a red 4 on slot 1 and a red 3 on slot 2.
  - Bob sees that this is a weird clue, because Alice has a *Free Choice* between cluing red and cluing 3. Alice is intentionally focusing the wrong card.
  - Bob blind-plays his *Finesse Position* as an *Undefined Move*. It is a blue 1 and it successfully plays on the stacks.
  - Cathy marks her slot 1 card as the red 4. (Cathy knows that Alice performed a *Bluff* and slot 1 was the focus of the clue.)

piano do you agree that the document does not specify this move, and this it is illegal?
by default, cathy should **never** assume that this is a move that has happened.
if you disagree, then we should get the move defined and added to the document.
this is an important building block for understanding what i am outlining here

**Comment by 0nen** [comment]
Level 10. The Bluff.

```
In this situation, players can perform an alternate strategy by fibbing to the next player. By cluing an unrelated, one-away-from-playable card, the next player will think that it is a Finesse and that they have the connecting card. Then, they will blind-play their Finesse Position card. This is called a Bluff.
```

It doesn't say you can't clue more than one card with a bluff. The focus was one away from playable, so it's a bluff. Simple.

**Comment by Zamiell** [comment]
> It doesn't say you can't clue more than one card with a bluff. The focus was one away from playable, so it's a bluff. Simple.

it's not simple at all. if bob played red 3, then red 3 would be duplicated, which would be illegal. thus, it isn't a bluff.

**Comment by timotree3** [comment]
In the hypothetical with red 2 on the stacks it's true that Bob does not believe it is a finesse, but that does not prevent it from being a bluff.
```
Known Bluffs

- Usually, when a Bluff occurs, the blind-playing player has no idea that is a Bluff - they assume they are playing a specific card into a true Finesse.
- However, in rare situations, a clue will be given that looks like a Finesse, but the next player will know for sure that they do not have the "connecting" card.
- In these situations, if the next player cannot see a better explanation for the clue, then they should blind-play their Finesse Position card as a Known Bluff.
```

**Comment by 0nen** [comment]
> > It doesn't say you can't clue more than one card with a bluff. The focus was one away from playable, so it's a bluff. Simple.
> 
> it's not simple at all. if bob played red 3, then red 3 would be duplicated, which would be illegal. thus, bob would not blind play.

Why can't it just be simply known that you are being bluffed and it would not duplicate? Or if there is duplication it would be known trash due to other cards on the board. We wouldn't bad touch if it can be avoided

**Comment by Zamiell** [comment]
> In the hypothetical with red 2 on the stacks it's true that Bob does not believe it is a finesse, but that does not prevent it from being a bluff.

agreed. but a known bluff in exactly this case is extremely rare. thus, cathy should not consider this possibility, and should write a note of exactly red 5 on her slot 2 card.

**Comment by Zamiell** [comment]
> Why can't it just be simply known that you are being bluffed and it would not duplicate?

are you talking about a known bluff?

**Comment by 0nen** [comment]
> > Why can't it just be simply known that you are being bluffed and it would not duplicate?
> 
> are you talking about a known bluff?

No, I'm saying it would either be a bluff, or if it duplicates then the other person knows their extra card is trash. So not known bluff.

**Comment by timotree3** [comment]
If you read the third bullet point in my Known Bluffs quote, it states that Bob should blind-play finesse position if there is not a better explanation for the clue. Therefore unless there is a better explanation for the clue, it is a perfectly conventional and defined clue as a known bluff. @Zamiell, do you assert that there is a better explanation for the clue in the hypo with r2 on the stacks? If so, which explanation, and why is it better than a known bluff?

**Comment by Zamiell** [comment]
> @Zamiell, do you assert that there is a better explanation for the clue in the hypo with r2 on the stacks? If so, which explanation, and why is it better than a known bluff?

Known Bluffs are more commonly performed when e.g.
- you are supposed to be playing r3, and you see one copy of r3 in the trash and the other copy of r3 in someone elses hand
- you are supposed to be playing r3 and the card you are blind-playing has negative red

but yes you are correct in that a known bluff technically could/should apply here

**Comment by Zamiell** [comment]
with that said, i maintain that:

> cathy should not consider this possibility, and should write a note of exactly red 5 on her slot 2 card. (this is historically how we have played)

**Comment by timotree3** [comment]
If a known bluff should apply, do you still assert that without a new convention of Faulty Bluffs, such clues are undefined?

**Comment by sjdrodge** [comment]
I think maybe the major confusion that you're having here Zamiel is how different the OOO Finesse scenario is from the scenario(s) we're talking about in this thread. With the OOO Finesse, there's real tension between interpreting it as a Known Bluff and an OOO Finesse, and I'm pretty sure the prescription is OOO Finesse by default. For example, Cathy has r3 b2 r1 y5 and Alice clues red to Cathy. Bob should think he's supposed to fix and play r2 later, not blind-play now and have r1 prompted later.

In the scenarios in this thread, Bob's alternatives are not blind-play now or blind-play later, Bob's alternatives are blind-play and no fix required, or give a fix and never blind-play. The latter is strongly counter-indicated when Alice could've just given a straightforward clue that didn't require a fix. Myself and others are shocked to hear that you believe the reasoning from the OOO Finesse scenario overrides the consideration that Alice shouldn't give lies for no reason.

**Comment by Zamiell** [comment]
> do you still assert that without a new convention of Faulty Bluffs, such clues are undefined?

no. allow me to change my position:

i'll modify the hypothetical:

- For example, in a 3-player game:
  - Red 2 is played on the stacks.
  - Alice clues red to Cathy, touching a red 4 on slot 1 and a red 3 on slot 2.
  - Bob sees that this is a weird clue, because Alice has a *Free Choice* between cluing red and cluing 3. Alice is intentionally focusing the wrong card.
  - Bob blind-plays his *Finesse Position* as a *Known Bluff*. It is a blue 1 and it successfully plays on the stacks.
  - Cathy marks her slot 1 card as the red 4. (Cathy knows that Alice performed a *Bluff* and slot 1 was the focus of the clue.)
  - Cathy is unsure whether to mark her slot 2 card as [red 5] or [red 3, red 5]. She considers both possibilities:
    - If there are blocking cards in her hand, Alice might clue red to Cathy intending for Bob to give a follow up OOO *Fix Clue*. If this was the case, then her slot 2 card is probably not red 3, or else Bob would have given a OOO *Fix Clue*. Thus it would only be a [red 5].
    - If there are no blocking cards in her hand, it is possible that Alice performed a *Known Bluff* by using *Free Choice*. If this was the case, then her slot 2 card could be a red 3. But it would also be possible to be a [red 3, red 5].
    - Alice knows that *Known Bluffs* using *Free Choice* are completely rare. As an unstated convention that is not in the document, Cathy knows that she should disregard this rare possibility, and instead just simply write a note of [red 5].

**Comment by timotree3** [comment]
I'm relieved by your new position. :smiley: It makes much more sense to me than the previous ones. I think it's actually an interesting question which note Cathy should write. I used to write that `r5` note in Cathy's position but I'm not sure if I still would.

**Comment by sjdrodge** [comment]
This hypothetical and the hypothetical offered in my comment at the top of the thread as essentially the same, to my mind. I would expect Cathy to maintain the full superposition unless context collapses it, just as I would expect Cathy to assume Good Touch Principle instead of marking the r4 as trash in the first example. That said, I would also be wary that a mistake/misunderstanding is possible in these scenarios since they are edge cases that don't come up a lot. I believe that is the way that most people currently play, and I'm pretty sure the responses in this thread support that belief.

**Comment by Zamiell** [comment]
closing this in favor of a new issue

**Comment by florian-5f3759df** [comment]
What I think might be a productive example to get to the disagreement:

No variant, r3, g1 played
Scenario A:
Bob: b1 - x - x - x
Cathy: r4 - g1 - g1 - r5
Alice clues red to Cathy. Since it seems much cleaner for Alice to clue 4 or 5 and let Alice clue red, Bob attempts to play the other copy of r4. 

Scenario B:
Bob: b1 - x - x - x
Cathy: r1 - g1 - g1 - r5
Alice clues red to Cathy. Bob plays b1, attempting to play r4. 

How I thought I understood the implications of the conventions, but apparently was wrong: Cathy can play the extra card by good touch. In Scenario A, this will work out well. In Scenario B, Cathy will bomb r1 as r4, so this was a clue that's illegal as a bluff, and the clue was only allowed if it was the truth.

What the consequences of the proposal sound like to me: Cathy is supposed to discard the extra red card, and therefore Scenario B is a legal bluff - Cathy will simply toss r1. The clue in Scenario A is illegal, Alice can only give this clue if it's the truth.

**Comment by 0nen** [comment]
> > do you still assert that without a new convention of Faulty Bluffs, such clues are undefined?
> 
> no. allow me to change my position:
> 
> i'll modify the hypothetical:
> 
> * For example, in a 3-player game:
>   
>   * Red 2 is played on the stacks.
>   * Alice clues red to Cathy, touching a red 4 on slot 1 and a red 3 on slot 2.
>   * Bob sees that this is a weird clue, because Alice has a _Free Choice_ between cluing red and cluing number 3. Bob sees that Alice is intentionally focusing the wrong card.
>   * Bob blind-plays his _Finesse Position_ as a _Known Bluff_. It is a blue 1 and it successfully plays on the stacks.


I know this is closed, but I just want to point out:

If Bob plays finesse it could still be r3. This would provide extra info: r5 is definitely in someone's hand, and Cathy has a known trash red card. Any player that doesn't see r5 would put elimination notes on their hand for it.

tl;dr It isn't 100% a Known Bluff, it's just very likely to be one.

**Comment by dobi815** [comment]
Any assumption to fix someone else's clue is dangerous. e.g. of bomb from just today https://hanab.live/replay/478480#4

The driver ofc. is the truth is visible in front of kopen's eyes and they are deciding between fixing MangoPie cleanly vs. blind playing but lanvin cannot tell the difference.

Fixing must be the exception and cannot be put ahead of blind playing. Its not how I play
